craft_application/git/_git_repo.py

```diff
@@ -308,3 +405,50 @@ def _resolve_ref(self, ref: str) -> str:
             ) from error
         logger.debug("Resolved reference %r for name %r", reference, ref)
         return reference
+
+    @classmethod
+    def clone_repository(
+        cls,
+        *,
+        url: str,
+        path: Path,
+        checkout_branch: str | None = None,
+        depth: int = 0,
+    ) -> GitRepo:
```

Comment:
```suggestion
    ) -> Self:
```

Reply:
Done in https://github.com/canonical/craft-application/pull/396/commits/7d210e4d46d63bf131a38f3f583a6aaa31202618